llvm/lib/SYCLLowerIR/LowerInvokeSimd.cpp

```diff
@@ -259,6 +262,49 @@ void markFunctionAsESIMD(Function *F) {
   }
 }
 
+void AdjustAddressSpace(Function *F, uint32_t ArgNo, uint32_t ArgAddrSpace) {
+  Argument *Arg = F->getArg(ArgNo);
+  for (User *ArgUse : Arg->users()) {
+    Instruction *Instr = dyn_cast<Instruction>(ArgUse);
+    if (!Instr)
+      continue;
+    const AddrSpaceCastInst *ASC = dyn_cast<AddrSpaceCastInst>(ArgUse);
+    if (ASC) {
+      if (ASC->getDestAddressSpace() == ArgAddrSpace)
+        continue;
+    }
+
+    const CallInst *CI = dyn_cast<CallInst>(ArgUse);
+    if (CI) {
+      Function *Callee = CI->getCalledFunction();
+      if (Callee) {
+        if (Callee->isDeclaration())
+          continue;
+      }
+
+      for (uint32_t i = 0; i < CI->getNumOperands(); ++i) {
+        if (CI->getOperand(i) == Arg) {
+          AdjustAddressSpace(Callee, i, ArgAddrSpace);
+        }
+      }
+    } else {
+      for (unsigned int i = 0; i < ArgUse->getNumOperands(); ++i) {
+        if (ArgUse->getOperand(i) == Arg) {
+          const Type *Ty = ArgUse->getOperand(i)->getType();
```

Comment:
It looks like we only use this variable to get the context, probably we could get rid of it and get the context from somewhere else